ouroboros-network/src/Ouroboros/Network/Mux.hs

```diff
@@ -57,27 +58,15 @@ muxStart udesc bearer rescb_m = do
                ]
     mjobs <- sequence [ mpsJob cnt pmss ptcl
                       | ptcl <- [minBound..maxBound] ]
-    aids <- mapM async $ jobs ++ concat mjobs
-    muxBearerSetState bearer Mature
-    watcher aids
 
-  where
-    watcher :: [Async m a] -> m ()
-    watcher as = do
-        (_,r) <- waitAnyCatchCancel as
-        close bearer
+    mask $ \unmask -> do
+      as <- traverse (async . unmask) (jobs ++ concat mjobs)
+      muxBearerSetState bearer Mature
+      unmask (void $ waitAny as) `onException` do
+        mapM_ cancel as
```

Comment:
This leaks threads if a mini protocol exits normally. waitAnyCancel works, e.g:
`unmask (void $ waitAnyCancel as)
muxBearerSetState bearer Dead